wizard.py

```diff
@@ -0,0 +1,1038 @@
+@with_session_id
+def versioning():
+  """Step 1.
+  Enter information about version control system. """
+  options = tooldb.collection["vcs"]
+
+  if request.method == "POST":
+    # Grab the form posted vcs commands and persist
+    # FIXME: Needs sanitizing
+    vcs_data = {
+      "items": [{"cmd": cmd} for cmd in request.form.getlist("vcs_cmd[]")],
+      "comment": request.form.get("comment", "")
+    }
+    _persist_session_subdocument_ts({"vcs": vcs_data})
+
+    flash("Now let's see how you build your software...",
+        "alert-success")
+    return redirect(url_for("building"))
+
+  user_data = _get_session_subdocument("vcs")
+  return render_template("versioning.html", options=options,
+      user_data=user_data)
+
+
+@app.route("/building", methods=["GET", "POST"])
+@with_session_id
+def building():
+  """Step 2.
+  Enter information about building. """
+  options = tooldb.collection["building"]
+
+  if request.method == "POST":
+    # Grab the form posted building commands and persist
+    # FIXME: Needs sanitizing
+    building_data = {
+      "items": [{"cmd": cmd} for cmd in request.form.getlist("build_cmd[]")],
+      "comment": request.form.get("comment", "")
+    }
+    _persist_session_subdocument_ts({"building": building_data})
+
+    flash("Let's talk about quality management next...",
+        "alert-success")
+    return redirect(url_for("quality_management"))
+
+  user_data = _get_session_subdocument("building")
+  return render_template("building.html", options=options, user_data=user_data)
+
+
+@app.route("/quality", methods=["GET", "POST"])
+@with_session_id
+def quality_management():
+  """Step 3.
+  Enter information about quality management. """
+  options = tooldb.collection["qa"]
+
+  if request.method == "POST":
+    # Grab the form posted quality management data and persist
+    # FIXME: Needs sanitizing
+    cmd_list = request.form.getlist("cmd[]")
+
+    retval_include_list = request.form.getlist("retval_include[]")
+    retval_operator_list = request.form.getlist("retval_operator[]")
+    retval_value_list = request.form.getlist("retval_value[]")
+
+    stdout_include_list = request.form.getlist("stdout_include[]")
+    stdout_operator_list = request.form.getlist("stdout_operator[]")
+    stdout_value_list = request.form.getlist("stdout_value[]")
+
+    stderr_include_list = request.form.getlist("stderr_include[]")
+    stderr_operator_list = request.form.getlist("stderr_operator[]")
+    stderr_value_list = request.form.getlist("stderr_value[]")
+
+    # Values of a step are related by the same index
+    # All lists must be equally long
+    # FIXME: Don't assert, try!
+    assert(len(cmd_list) ==
+        len(retval_include_list) == len(retval_operator_list) ==
+        len(retval_value_list) == len(stdout_include_list) ==
+        len(stdout_operator_list) == len(stdout_value_list) ==
+        len(stderr_include_list) == len(stderr_operator_list) ==
+        len(stderr_value_list))
```

Comment:
hmm, can't this assertion be moved to a form object?

Reply:
Good idea, while Flask does not have built-in form objects I could make more use of`Flask-WTF`, which I recently added for CSRF protection.

Reply:
Yeah, I think this is what we use in the arch security tracker.

Reply:
Will fix with https://github.com/in-toto/layout-web-tool/issues/5